File: powerOfThree.py
# problem: https://leetcode.com/problems/power-of-three/
# Runtime: 64 ms, faster than 93.23% of Python3 online submissions for Power of Three.
# Memory Usage: 14.1 MB, less than 91.80% of Python3 online submissions for Power of Three.

import math


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        return n > 0 and (math.log10(n) / math.log10(3)) % 1 == 0

File: powerOfThree.cpp
// problem: https://leetcode.com/problems/power-of-three/submissions/
// Runtime: 39 ms, faster than 38.01% of C++ online submissions for Power of
// Three. Memory Usage: 5.9 MB, less than 29.00% of C++ online submissions for
// Power of Three.
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  bool isPowerOfThree(int n) {
    if (n <= 0)
      return false;
    while (n % 3 == 0)
      n /= 3;
    return n == 1;
  }
};
